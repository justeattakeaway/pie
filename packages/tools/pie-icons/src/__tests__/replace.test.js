import replace from '../replace';

vitest.mock('../../dist/icons.json', () => ({
    default: {
        icon1: {
            contents:
        '<line x1="23" y1="1" x2="1" y2="23" /><line x1="1" y1="1" x2="23" y2="23" />',
        },
        icon2: {
            contents: '<circle cx="12" cy="12" r="11" />',
        },
    },
}));

test('replaces [data-pie-icons] elements with SVG markup', () => {
    document.body.innerHTML =
    '<i data-pie-icons="icon1"></i><span data-pie-icons="icon2"></i>';
    expect(document.body.innerHTML).toMatchSnapshot();
    replace();
    expect(document.body.innerHTML).toMatchSnapshot();
});

test('copies placeholder element attributes to <svg> Tag', () => {
    document.body.innerHTML =
    '<i data-pie-icons="icon1" id="test" class="foo bar" stroke-width="1"></i>';
    expect(document.body.innerHTML).toMatchSnapshot();
    replace();
    expect(document.body.innerHTML).toMatchSnapshot();
});

test('sets attributes passed as parameters', () => {
    document.body.innerHTML =
    '<i data-pie-icons="icon1" id="test" class="foo bar" stroke-width="1"></i>';
    expect(document.body.innerHTML).toMatchSnapshot();
    replace({ class: 'foo bar hello', fill: 'salmon', 'stroke-width': 1.5 });
    expect(document.body.innerHTML).toMatchSnapshot();
});
