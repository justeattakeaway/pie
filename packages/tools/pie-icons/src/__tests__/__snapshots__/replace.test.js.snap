// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copies placeholder element attributes to <svg> Tag 1`] = `"<i data-pie-icons=\\"icon1\\" id=\\"test\\" class=\\"foo bar\\" stroke-width=\\"1\\"></i>"`;

exports[`copies placeholder element attributes to <svg> Tag 2`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" id=\\"test\\" class=\\"c-pieIcon c-pieIcon--icon1 foo bar\\" stroke-width=\\"1\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\"></line><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\"></line></svg>"`;

exports[`replaces [data-pie-icons] elements with SVG markup 1`] = `"<i data-pie-icons=\\"icon1\\"></i><span data-pie-icons=\\"icon2\\"></span>"`;

exports[`replaces [data-pie-icons] elements with SVG markup 2`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" class=\\"c-pieIcon c-pieIcon--icon1\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\"></line><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\"></line></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" class=\\"c-pieIcon c-pieIcon--icon2\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"11\\"></circle></svg>"`;

exports[`sets attributes passed as parameters 1`] = `"<i data-pie-icons=\\"icon1\\" id=\\"test\\" class=\\"foo bar\\" stroke-width=\\"1\\"></i>"`;

exports[`sets attributes passed as parameters 2`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"presentation\\" focusable=\\"false\\" class=\\"c-pieIcon c-pieIcon--icon1 foo bar hello\\" color=\\"salmon\\" stroke-width=\\"1\\" id=\\"test\\"><line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\"></line><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\"></line></svg>"`;
